src/core/providers/ogr/qgsogrconnpool.cpp

```diff
@@ -18,6 +18,43 @@
 
 #include "qgslogger.h"
 
+
+void QgsOgrConnPoolGroup::connectionCreate( const QString &connInfo, QgsOgrConn *&c )
+{
+  c = new QgsOgrConn;
+
+  const QVariantMap parts = QgsOgrProviderMetadata().decodeUri( connInfo );
+  const QString fullPath = parts.value( QStringLiteral( "vsiPrefix" ) ).toString()
+                           + parts.value( QStringLiteral( "path" ) ).toString()
+                           + parts.value( QStringLiteral( "vsiSuffix" ) ).toString();
+  const QStringList openOptions = parts.value( QStringLiteral( "openOptions" ) ).toStringList();
+  char **papszOpenOptions = nullptr;
+  for ( const QString &option : openOptions )
```

Comment:
```suggestion
  for ( const QString &option : std::as_const( openOptions ) )
```

I know it's redundant, but it makes it explicit and it's a noop anyway.